### PR TITLE
Add interpreter for ShiftRightArithmeticOp

### LINKED ISSUE #1113: Add interpreter for ShiftRightArithmeticOp
Following the [testing guidelines](https://github.com/openxla/stablehlo/blob/main/docs/interpreter_checklist.md)

### PR DESCRIPTION
**Pull request by ghpvnist**
Here are the constraints for the ShiftRightArithmeticOp:
```
(I1) lhs is a tensor of integer type.
(I2) rhs is a tensor of integer type.
(C1) `lhs`, `rhs`, and `result` have the same type.
```
I1, I2, and C1 are covered by the ODS, so no additional tests are added.

Notes:
* Corner cases (shift overflow) has not been accounted for: #1150

closes #1113

### CODE REVIEW

File: docs/spec.md [hunk 1]
```diff
@@ -4977,12 +4977,14 @@ Performs element-wise arithmetic right-shift operation on the `lhs` tensor by
 #### Examples
 
 ```mlir
-// %lhs: [-1, -128, -36, 5, 3, 7]
-// %rhs: [1, 2, 3, 2, 1, 3]
-%result = "stablehlo.shift_right_arithmetic"(%lhs, %rhs): (tensor<6xi8>, tensor<6xi8>) -> tensor<6xi8>
-// %result: [-1, -32, -5, 1, 1, 0]
+// %lhs: [-1, 0, 1]
+// %rhs: [1, 2, 3]
+%result = "stablehlo.shift_right_arithmetic"(%lhs, %rhs): (tensor<3xi64>, tensor<3xi64>) -> tensor<3xi64>
```
Comment (sdasgup3) [inline docs/spec.md:4982]: The input `lhs = 1` and `rhs = 3` add some value for left shift. For right shift, it would be better to have an input like shiftArithRight(16, 3) = 1, to exemplify the function. This would also include changing the ODS and interpreter tests.

File: stablehlo/dialect/StablehloOps.td [hunk 2]
```diff
@@ -883,7 +883,8 @@ def StableHLO_ShiftRightArithmeticOp : StableHLO_BinaryElementwiseOp<"shift_righ
 }
 
 def StableHLO_ShiftRightLogicalOp : StableHLO_BinaryElementwiseOp<"shift_right_logical",
-      [Pure, HLO_CompatibleOperandsAndResultType], HLO_IntTensor> {
+      [Pure, HLO_CompatibleOperandsAndResultType /*shift_right_logical_c1*/],
+      HLO_IntTensor /*shift_right_logical_i1, shift_right_logical_i2*/> { /*shift_right_logical_c1*/
```
Comment (sdasgup3) [inline stablehlo/dialect/StablehloOps.td:887]: logical -> arithmetic

File: stablehlo/dialect/StablehloOps.td [hunk 3]
```diff
@@ -894,7 +895,7 @@ def StableHLO_ShiftRightLogicalOp : StableHLO_BinaryElementwiseOp<"shift_right_l
 
     Example:
     ```mlir
-    %result = stablehlo.shift_right_logical %lhs, %rhs : tensor<6xi8>
+    %result = stablehlo.shift_right_logical %lhs, %rhs : tensor<3xi64>
```
Comment (sdasgup3) [inline stablehlo/dialect/StablehloOps.td:898]: logical -> arithmetic

File: stablehlo/dialect/StablehloOps.td [hunk 4]
```diff
@@ -883,7 +883,8 @@ def StableHLO_ShiftRightArithmeticOp : StableHLO_BinaryElementwiseOp<"shift_righ
 }
 
 def StableHLO_ShiftRightLogicalOp : StableHLO_BinaryElementwiseOp<"shift_right_logical",
```
Comment (sdasgup3) [inline stablehlo/dialect/StablehloOps.td:885]: wrong ODS def updated?

**Review by sdasgup3 — APPROVED** [review]
